lib/FrontendTool/ReferenceDependencies.cpp

```diff
@@ -102,6 +102,17 @@ static std::string mangleTypeAsContext(const NominalTypeDecl *type) {
   return Mangler.mangleTypeAsContextUSR(type);
 }
 
+std::vector<std::string>
+swift::reversePathSortedFilenames(const ArrayRef<std::string> elts) {
```

Comment:
Not an essential part of this change, but as I'm building scaffolding to catch dependency-related bugs like the one that motivated this (and other bugs in incremental mode) elsewhere, the sorting change makes the files emitted from dependency analysis less dependent on the order in which clang and/or swift notice them. Currently they're unstable, so an incremental build will emit a different set of external dependencies depending on which files it touches (and different set from a full build). Figured I'd take the opportunity to tidy it up.

As for the weird order: it's to make the order as stable and independent of path prefix variation as possible. If we sort by normal string order, path differences between where system libraries, user libraries, the testing workspace and so forth will figure into the order (since those occur as prefixes). It's not ideal but it seemed the best I could manage.